platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java

```diff
@@ -1,11 +1,56 @@
 package uno.cod.platform.server.codingcontest.sync.service;
 
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uno.cod.platform.server.codingcontest.sync.dto.CodingcontestDto;
 import uno.cod.platform.server.codingcontest.sync.dto.ParticipationDto;
+import uno.cod.platform.server.core.domain.Challenge;
+import uno.cod.platform.server.core.domain.User;
+import uno.cod.platform.server.core.repository.ChallengeRepository;
+import uno.cod.platform.server.core.repository.UserRepository;
 
+import javax.transaction.Transactional;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@Service
+@Transactional
 public class CodingcontestSyncService {
+    private final UserRepository userRepository;
+    private final ChallengeRepository challengeRepository;
+
+    @Autowired
+    public CodingcontestSyncService(UserRepository userRepository, ChallengeRepository challengeRepository) {
+        this.userRepository = userRepository;
+        this.challengeRepository = challengeRepository;
+    }
 
-    public void updateUserFromCodingcontest(ParticipationDto dto) {
+    public User updateUserFromCodingcontest(ParticipationDto dto) {
+        return userRepository.save(mapParticipationDto(dto));
+    }
+
+    public void createContest(CodingcontestDto dto){
+        Challenge challenge = new Challenge();
```

Comment:
you should not create a new challenge everytime the button gets hit, this is also used for updating the contest